src/9on12Resource.cpp

```diff
@@ -1381,7 +1383,78 @@ namespace D3D9on12
         }
 
         mapType = GetMapTypeFlag(flags.ReadOnly, flags.WriteOnly, flags.Discard, flags.NoOverwrite, m_isDecodeCompressedBuffer, m_pResource->AppDesc()->CPUAccessFlags());
-        
+
+        if (RegistryConstants::g_cLockDiscardOptimization)
+        {
+            if (mapType == D3D12TranslationLayer::MAP_TYPE_WRITE_DISCARD)
+            {
+                auto lockedResourceRanges = device.m_lockedResourceRanges.GetLocked();
+
+                // Check if buffer is in the map
+                // If it is, get a list of previously mapped ranges
+                if (lockedResourceRanges->count(this))

```

Comment:
Instead of using `count()` and `at()`, this should be a `find()` so that the container only needs to be searched once, and then the iterator can be dereferenced instead of using `at()` to re-search the container over and over again.